lib/Logger/Logger.cpp

```diff
@@ -550,29 +561,37 @@ void Logger::log(char const* logid, char const* function, char const* file, int
 
   auto msg = std::make_unique<LogMessage>(function, file, line, level, topicId, std::move(out), offset);
 
+  append(defaultLogGroup(), msg, [level, topicId](std::unique_ptr<LogMessage>& msg) -> void {
+    LogAppenderStdStream::writeLogMessage(STDERR_FILENO, (isatty(STDERR_FILENO) == 1),
+                                          level, topicId, msg->_message.data(),
+                                          msg->_message.size(), true);
+  });
+} catch (...) {
+  // logging itself must never cause an exeption to escape
+}
+
+void Logger::append(LogGroup& group, std::unique_ptr<LogMessage>& msg,
+                    std::function<void(std::unique_ptr<LogMessage>&)> inactive) {
```

Comment:
This copying the std::function every time.
```suggestion
                    std::function<void(std::unique_ptr<LogMessage>&)> const& inactive) {
```